library/Director/Health.php

```diff
@@ -36,11 +36,17 @@ public function setDbResourceName($name)
         return $this;
     }
 
-    public function getCheck($name)
+    public function getCheck($name,$check_name = null)
     {
         if (array_key_exists($name, $this->checks)) {
-            $func = $this->checks[$name];
-            $check = $this->$func();
+            if(!$check_name){
```

Comment:
A strict `null` check is absolutely necessary if you really want to compare with the specified default value of `null`.  I would also swap the `if` branches again,  i.e. `if ($checkName !== null) { ...`.